### DISCUSSION TITLE
为什么seata控制台报了回滚成功，实际上还是无效

### DISCUSSION THREAD

**Discussion by gitb1ock** [apache/incubator-seata#4525] [Q&A] [answered]
![I$WJ%~9EY XHJWB3NX2E9PM](https://user-images.githubusercontent.com/90310205/161782379-f73ca88e-fb43-43f9-aa3f-40bf596f5192.png)

**Comment by pengten** [comment]
检查分支有没注册成功

**Reply by gitb1ock** [reply]
![image](https://user-images.githubusercontent.com/90310205/162117562-9b831fdb-3358-4b38-8002-0f3e2b61ea19.png)

这是抛出异常的被调用方的日志，没有出现branch rollback，能说明分支没有注册成功吗？如果能，请问该怎么解决

**Reply by pengten** [reply]
去服务端日志看，有没有对应分支的Register branch successfully的日志，没有就说明没注册成功。一般是因为XID没有传递，有没有使用spring-cloud-alibaba-seata？

**Reply by gitb1ock** [reply]
没有Register branch successfully的日志
![WSL$PKWL7~Q5WP83~W@0RZ4](https://user-images.githubusercontent.com/90310205/162437041-e2e5ea01-6159-44cf-b1cb-13503518ba97.png)
这是使用的依赖

**Comment by funky-eyes** [comment, marked as answer]
检查xid传递和数据源代理

**Reply by gitb1ock** [reply]
确实是数据源代理出了问题，谢谢
